tests/MongoDB.Driver.Core.Tests/Core/Operations/BulkMixedWriteOperationTests.cs

```diff
@@ -198,6 +198,42 @@ public void WriteConcern_should_work()
             }
         }
 
+        [Theory]
+        [ParameterAttributeData]
+        public void Execute_with_hint_should_throw_when_hint_is_not_supported(
+            [Values(false, true)] bool async)
+        {
+            var requests = new List<WriteRequest>
+            {
+                new UpdateRequest(
+                    UpdateType.Update,
+                    new BsonDocument("x", 1),
+                    new BsonDocument("$set", new BsonDocument("x", 2)))
+                {
+                    Hint = new BsonDocument("_id", 1)
+                }
+            };
+            var subject = new BulkMixedWriteOperation(_collectionNamespace, requests, _messageEncoderSettings);
+
+            var exception = Record.Exception(() => ExecuteOperation(subject, async));
+
+            if (!Feature.HintForUpdateAndReplaceOperations.IsSupported(CoreTestConfiguration.ServerVersion, out var isExceptionAllowed))
+            {
+                if (isExceptionAllowed)
```

Comment:
Lines 220-223 cane become:

            var hintFeature = Feature.HintForUpdateAndReplaceOperations;
            var serverVersion = CoreTestConfiguration.ServerVersion;
            if (!hintFeature.IsSupported(serverVersion))
            {
                if (hintFeature.DriverMustThrowIfNotSupported(serverVersion))

Reply:
I've altered the if flow a bit to include the case when hint is supported.
@rstam Could you please confirm if you are ok with this change?

Reply:
Yes it's OK.